component/LinkableText/LinkableText.tsx

```diff
@@ -0,0 +1,25 @@
+import Link from "next/link";
+
+const LinkableText = ({text}: {text: string}) => {
+    const sanitizeNewLine = text?.split('\n')
+    const regexUrl = /^((?:https?:\/\/)?[^./]+(?:\.[^./]+)+(?:\/.*)?)$/
+
+    const textComponent = sanitizeNewLine?.map((textline, index) => {
+        const textLineSplit = textline?.split(' ')
+        return <p key={textline}>
+            {textLineSplit?.map((word, wordIndex) => {
+                if(wordIndex < textLineSplit?.length - 1) word += ' '
+                const hrefUrl = (word?.includes('http') || word?.includes('https')) ? word : `https://${word}`
+                if(regexUrl?.test(word)) return <Link className="underline text-signedPrimary" href={hrefUrl}>{word}</Link>
+                return word
+            })}
+        </p>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for appending a space to the word if it's not the last word in the line can be simplified by using `join(' ')` after mapping over `textLineSplit`. Also, the URL construction logic can be improved. Currently, it prepends 'https://' to every non-http(s) word which might not always be a valid URL.

```diff
-        const textLineSplit = textline?.split(' ')
-        return <p key={textline}>
-            {textLineSplit?.map((word, wordIndex) => {
-                if(wordIndex < textLineSplit?.length - 1) word += ' '
-                const hrefUrl = (word?.includes('http') || word?.includes('https')) ? word : `https://${word}`
-                if(regexUrl?.test(word)) return <Link className="underline text-signedPrimary" href={hrefUrl}>{word}</Link>
-                return word
-            })}
-        </p>

+        const textLineSplit = textline.split(' ')
+        return (
+          <p key={textline}>
+            {textLineSplit.map((word) => {
+              const hrefUrl = word.startsWith('http://') || word.startsWith('https://') ? word : `https://${word}`;
+              return regexUrl.test(hrefUrl) ? <Link className="underline text-signedPrimary" href={hrefUrl}>{word}</Link> : word;
+            }).join(' ')}
+          </p>
+        );
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->